build/commands/lib/config.js

```diff
@@ -1163,14 +1175,19 @@ Config.prototype.update = function (options) {
 
   if (options.mac_signing_identifier)
     this.mac_signing_identifier = options.mac_signing_identifier
+  else if (this.extraGnArgs['mac_signing_identifier'])
+    // We reach here during `npm run build -- --target=...sign_updater ...`.
+    this.mac_signing_identifier = this.extraGnArgs['mac_signing_identifier']
 
   if (options.mac_installer_signing_identifier)
     this.mac_installer_signing_identifier = options.mac_installer_signing_identifier
 
   if (options.mac_signing_keychain)
     this.mac_signing_keychain = options.mac_signing_keychain
 
-  if (options.notarize)
+  // We pass --gn notarize:true to `npm run build -- --target=...sign_updater`.
+  // So we need to check `extraGnArgs` here as well.
+  if (options.notarize || this.extraGnArgs['notarize'] === true)
```

Comment:
I don't think we should configure `build/create_dist` commands by passing `--gn` args directly. It's fine for local development, but it lacks basic validation to be used in production workflows (you can write `npm run build --gn helloworld:true`), and it feels like we're accessing some internals instead of using the supported configuration parameters `npm run build` offers.

Instead consider adding `--notarize` flag to `npm run build`. All these changes should not be needed (`--mac_signing_identifier` is supported by `npm run build`).

sidenote: I'm not sure why we have `create_dist` in the first place. It's just `npm run build` with a specific target and some additional parameters. Maybe we should remove it to simplify things.

Reply:
What do you think about the above @bridiver?

Reply:
> the canonical source of truth are the GN args
I agree with @goodov here, the GN args are the result of the commands we run. We have so many different defaults than upstream that it would be totally impractical to manage things via gn args directly which is why we have these helpers in the first place

Reply:
I switched to an implementation based on NPM args in 1a895c453455b75bc3f9c5c45e1ea18b2bbd9d01.